dbms/src/Core/Types.h

```diff
@@ -240,4 +261,14 @@ namespace std
                 ^ std::hash<DB::Int64>()(x.value & std::numeric_limits<DB::UInt64>::max());
         }
     };
+
+    template <>
+    struct hash<DB::Decimal256>
+    {
+        size_t operator()(const DB::Decimal256 & x) const
+        {
+            return std::hash<DB::Int128>()(x.value >> 128)
```

Comment:
This is a classic mistake.

Please watch this video: https://www.youtube.com/watch?v=EoX82TEz2sQ
https://presentations.clickhouse.tech/hash_tables/#11